apps/prairielearn/src/ee/institution/utils.ts

```diff
@@ -1,38 +1,49 @@
-import {
-  loadSqlEquiv,
-  queryAsync,
-  queryOneRowAsync,
-  queryZeroOrOneRowAsync,
-} from '@prairielearn/postgres';
+import { loadSqlEquiv, queryOptionalRow, queryRow, queryRows } from '@prairielearn/postgres';
 
 import { config } from '../../lib/config';
+import {
+  AuthnProviderSchema,
+  InstitutionSchema,
+  SamlProviderSchema,
+  type AuthnProvider,
+  type Institution,
+  type SamlProvider,
+} from '../../lib/db-types';
 
 const sql = loadSqlEquiv(__filename);
 
-export async function getInstitution(institutionId) {
-  const institutionRes = await queryOneRowAsync(sql.select_institution, {
-    id: institutionId,
-  });
-  return institutionRes.rows[0];
+export async function getInstitution(institutionId: string): Promise<Institution> {
+  const institutionRes = await queryRow(
+    sql.select_institution,
+    { id: institutionId },
+    InstitutionSchema
+  );
+  return institutionRes;
 }
 
-export async function getInstitutionSamlProvider(institutionId) {
-  const samlProviderRes = await queryZeroOrOneRowAsync(sql.select_institution_saml_provider, {
-    institution_id: institutionId,
-  });
-  return samlProviderRes.rows[0] ?? null;
+export async function getInstitutionSamlProvider(
+  institutionId: string
+): Promise<SamlProvider | null> {
+  return await queryOptionalRow(
+    sql.select_institution_saml_provider,
+    { institution_id: institutionId },
+    SamlProviderSchema
+  );
 }
 
-export async function getInstitutionAuthenticationProviders(institutionId) {
-  const authProvidersRes = await queryAsync(sql.select_institution_authn_providers, {
-    institution_id: institutionId,
-  });
-  return authProvidersRes.rows;
+export async function getInstitutionAuthenticationProviders(
+  institutionId: string
+): Promise<AuthnProvider[]> {
+  return await queryRows(
+    sql.select_institution_authn_providers,
+    { institution_id: institutionId },
+    AuthnProviderSchema
+  );
 }
 
-export async function getSupportedAuthenticationProviders() {
-  const authProvidersRes = await queryAsync(sql.select_authentication_providers, {});
-  return authProvidersRes.rows.filter((row) => {
+export async function getSupportedAuthenticationProviders(): Promise<AuthnProvider[]> {
+  const authProviders = await queryRows(sql.select_authentication_providers, AuthnProviderSchema);
```

Comment:
I would call this variable the plural table name.
```suggestion
  const auth_providers = await queryRows(sql.select_authentication_providers, AuthnProviderSchema);
```
As the variable names become more complex I'm less sure what to do. E.g., if we have new and old authn providers, are they `new_authn_provider`/`old_authn_provider` or `newAuthnProvider`/`oldAuthnProvider`? I haven't always been consistent with this.